source/common/ssl/connection_impl.cc

```diff
@@ -139,16 +135,25 @@ Network::ConnectionImpl::PostIoAction ConnectionImpl::doHandshake() {
 
 void ConnectionImpl::drainErrorQueue() {
   bool saw_error = false;
+  bool saw_counted_error = false;
   while (uint64_t err = ERR_get_error()) {
     if (!saw_error) {
```

Comment:
Good catch, I've been micro-optimizing too much ;)

@mattklein123 any thoughts on style? We can either leave it as is (i.e. `if (!saw_error) { saw_error = true; }`) or always set it unconditionally (i.e. `saw_error = true;`), like suggested by @myidpt. Both are fine with me.